src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2882,7 +2882,17 @@ private uint getJitFlags(ref CORJIT_FLAGS flags, uint sizeInBytes)
             }
 
             if (this.MethodBeingCompiled.IsNativeCallable)
+            {
+#if READYTORUN
+                if (targetArchitecture == TargetArchitecture.X86
+                    && _compilation.TypeSystemContext.Target.OperatingSystem == TargetOS.Windows)
+                {
```

Comment:
Could you please add equivalent logic for crossgen1?

E.g. to `ZapInfo::CompileMethod`:

```
#if defined(TARGET_WINDOWS) && (TARGET_X86)
    if (GetCompileInfo()->IsNativeCallableMethod(m_currentMethodHandle))

```